### PR TITLE
DOC: added docstring for define_app along with other docs

### PR DESCRIPTION
**Pull request by GavinHuttley**
[NEW] hooked up API docs

[NEW] updated app docs to new usage of define_app for decorating
    user functions.

### CODE REVIEW

File: doc/app/user_function.rst [hunk 1]
```diff
@@ -1,48 +1,33 @@
 Turn your functions into composable apps
 ========================================
 
-This is super easy -- just use the ``appify`` decorator! This generates a ``user_function`` wrapper class that takes a reference to your function, its input, output, and data types. The resulting app can then become part of a composed function.
+This is super easy -- just use the ``define_app`` decorator! This generates a wrapper class that has a reference to your functionand can then become part of a composed function.
```
Comment (Nick-Foto) [inline doc/app/user_function.rst:4]: function and

Comment (Nick-Foto) [reply doc/app/user_function.rst:4]: becomes part of a composed class.

File: doc/app/user_function.rst [hunk 2]
```diff
@@ -83,22 +68,17 @@ This time we wrap a method call on a ``SequenceCollection`` (and the alignment s
 A user app with a different output type
 ---------------------------------------
 
-In this example, we make an function that returns ``DistanceMatrix`` of an alignment.
+In this example, we make an function that returns a ``DistanceMatrix`` from an alignment.
```
Comment (Nick-Foto) [inline doc/app/user_function.rst:71]: a function

File: doc/app/user_function.rst (hunk 1 above)
Comment (GavinHuttley) [reply doc/app/user_function.rst:4]: Good catch!

**Pull request merged**